### PR TITLE
Integrates circles with Parse

### PR DESCRIPTION
**Pull request by jcomo**
### Overview

Adds the ability to save circles to Parse and retrieve them. Introduces the repository pattern for data access.
### TODO
- Need to persist images to parse. Users can select images for the circle, but can't save them yet
- Scope the query for friends to the user
### Reviewers
- [ ] @hashamali 
- [x] @dbuscaglia

### CODE REVIEW

File: app/src/main/java/it/quip/android/actvitiy/CreateCircleActivity.java [hunk 1]
```diff
@@ -66,6 +73,18 @@ public boolean onOptionsItemSelected(MenuItem item) {
 
     private void createCircle() {
         Circle createdCircle = circleHeaderFragment.getCircle();
+        createdCircle.saveInBackground(new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                Log.i("DEBUG", "save completed");
```
Comment (hashamali) [inline app/src/main/java/it/quip/android/actvitiy/CreateCircleActivity.java]: #thangs

File: app/src/main/java/it/quip/android/actvitiy/QuipitHomeActivity.java [hunk 2]
```diff
@@ -190,9 +202,9 @@ private void createQuip() {
     }
 
     private void onCircleCreated(Circle createdCircle) {
-        user.addCircle(createdCircle);
+        circles.add(createdCircle);
         updateSidebarMenu();
-        prepareFragment(ViewCircleFragment.newInstance(createdCircle)).commitAllowingStateLoss();
+        // prepareFragment(ViewCircleFragment.newInstance(createdCircle)).commitAllowingStateLoss();
```
Comment (dbuscaglia) [inline app/src/main/java/it/quip/android/actvitiy/QuipitHomeActivity.java]: somethangs

Comment (hashamali) [inline app/src/main/java/it/quip/android/actvitiy/QuipitHomeActivity.java]: what are them thangs

File: app/src/main/java/it/quip/android/fragment/InviteFriendsFragment.java [hunk 3]
```diff
@@ -126,10 +148,10 @@ public void afterTextChanged(Editable s) {
     private void searchFriends(String name) {
         aFilteredFriends.clear();
 
-        for (User friend : MockUtils.getUsers()) {
+        for (User friend : friends) {
```
Comment (dbuscaglia) [inline app/src/main/java/it/quip/android/fragment/InviteFriendsFragment.java]: Where are friends coming from

File: app/src/main/java/it/quip/android/repository/circle/ParseCircleRepository.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                        }
+
+                        handler.onSuccess(circles);
+                    }
+                });
+    }
+
+    @Override
+    public void save(Circle circle) {
+
```
Comment (dbuscaglia) [inline app/src/main/java/it/quip/android/repository/circle/ParseCircleRepository.java]: this is a pretty useless override

**Comment by dbuscaglia** [comment]
LGTM

**Pull request merged**